streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -826,18 +826,21 @@ long runOnce(final long recordsProcessedBeforeCommit) {
             }
         }
 
-        if (records != null && !records.isEmpty() && taskManager.hasActiveRunningTasks()) {
+        if (records != null && !records.isEmpty()) {
             streamsMetrics.pollTimeSensor.record(computeLatency(), timerStartedMs);
             addRecordsToTasks(records);
+        }
+
+        if (taskManager.hasActiveRunningTasks()) {
```

Comment:
why this change?

Reply:
As in description :)

```
Break the main loop on put-raw-data and process-them. Since now not all data put into the queue would be processed completely within a single iteration.
```